### PR TITLE
Use new staking transaction format

### PR DESCRIPTION
**Pull request by bowenwang1996**
Nearcore changed the format of staking transaction and this PR matches the schema change. However, the change is not backward compatible.

### CODE REVIEW

**Comment by vgrichina** [comment]
Need to add basic backwards compatibility support for previous version of nearcore (crash on staking attempt).

File: src.ts/account.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Action, transfer, createAccount, signTransaction, deployContract,
 import { FinalExecutionOutcome, FinalExecutionStatusBasic } from './providers/provider';
 import { Connection } from './connection';
 import {base_decode, base_encode} from './utils/serialize';
-import { PublicKey } from './utils/key_pair';
+import {BlsPublicKey, PublicKey} from './utils/key_pair';
```
Comment (janedegtiareva) [inline src.ts/account.ts]: Do we have docs somewhere on what's the diff between BlsPublicKey and regular PublicKey?

**Comment by vgrichina** [comment]
Needs to add tests for staking with BLS key as well (there are no tests for staking whatsoever)

File: src.ts/account.ts (hunk 1 above)
Comment (bowenwang1996) [reply src.ts/account.ts]: It's not really used in nearlib. Rather it just holds data without actually doing anything.

File: src.ts/account.ts [hunk 2]
```diff
@@ -173,8 +173,12 @@ export class Account {
         return this.signAndSendTransaction(this.accountId, [deleteKey(PublicKey.from(publicKey))]);
     }
 
-    async stake(publicKey: string | PublicKey, amount: BN): Promise<FinalExecutionOutcome> {
-        return this.signAndSendTransaction(this.accountId, [stake(amount, PublicKey.from(publicKey))]);
+    async stake(publicKey: string | BlsPublicKey, amount: BN): Promise<FinalExecutionOutcome> {
+        const status = await this.connection.provider.status();
+        if (status.version.version < '0.4.0') {
```
Comment (vgrichina) [inline src.ts/account.ts]: this isn't very robust way to compare versions, makes sense to use something like https://github.com/omichelsen/compare-versions

**Comment by bowenwang1996** [comment]
We removed bls in nearcore so this is no longer necessary.

**Pull request closed without merging**